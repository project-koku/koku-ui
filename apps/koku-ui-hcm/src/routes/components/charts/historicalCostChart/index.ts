export { default as HistoricalCostChart } from './historicalCostChart';
export type { HistoricalCostChartProps } from './historicalCostChart';
