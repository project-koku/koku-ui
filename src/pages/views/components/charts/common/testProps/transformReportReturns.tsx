const transformReportReturns = [
  {
    x: 1,
    y: 27,
    key: '2022-05-01',
    name: '2022-05-01',
    units: 'Core-Hours',
  },
  {
    x: 2,
    y: 69,
    key: '2022-05-02',
    name: '2022-05-02',
    units: 'Core-Hours',
  },
  {
    x: 3,
    y: 69,
    key: '2022-05-03',
    name: '2022-05-03',
    units: 'Core-Hours',
  },
  {
    x: 4,
    y: 69,
    key: '2022-05-04',
    name: '2022-05-04',
    units: 'Core-Hours',
  },
  {
    x: 5,
    y: 69,
    key: '2022-05-05',
    name: '2022-05-05',
    units: 'Core-Hours',
  },
  {
    x: 6,
    y: 69,
    key: '2022-05-06',
    name: '2022-05-06',
    units: 'Core-Hours',
  },
  {
    x: 7,
    y: 69,
    key: '2022-05-07',
    name: '2022-05-07',
    units: 'Core-Hours',
  },
  {
    x: 8,
    y: 69,
    key: '2022-05-08',
    name: '2022-05-08',
    units: 'Core-Hours',
  },
  {
    x: 9,
    y: 69,
    key: '2022-05-09',
    name: '2022-05-09',
    units: 'Core-Hours',
  },
  {
    x: 10,
    y: 69,
    key: '2022-05-10',
    name: '2022-05-10',
    units: 'Core-Hours',
  },
  {
    x: 11,
    y: 69,
    key: '2022-05-11',
    name: '2022-05-11',
    units: 'Core-Hours',
  },
  {
    x: 12,
    y: 78,
    key: '2022-05-12',
    name: '2022-05-12',
    units: 'Core-Hours',
  },
  {
    x: 13,
    y: 92,
    key: '2022-05-13',
    name: '2022-05-13',
    units: 'Core-Hours',
  },
  {
    x: 14,
    y: 92,
    key: '2022-05-14',
    name: '2022-05-14',
    units: 'Core-Hours',
  },
  {
    x: 15,
    y: 92,
    key: '2022-05-15',
    name: '2022-05-15',
    units: 'Core-Hours',
  },
  {
    x: 16,
    y: 92,
    key: '2022-05-16',
    name: '2022-05-16',
    units: 'Core-Hours',
  },
  {
    x: 17,
    y: 92,
    key: '2022-05-17',
    name: '2022-05-17',
    units: 'Core-Hours',
  },
  {
    x: 18,
    y: 92,
    key: '2022-05-18',
    name: '2022-05-18',
    units: 'Core-Hours',
  },
  {
    x: 19,
    y: 92,
    key: '2022-05-19',
    name: '2022-05-19',
    units: 'Core-Hours',
  },
  {
    x: 20,
    y: null,
    key: '2022-05-20',
    name: '2022-05-20',
    units: undefined,
  },
  {
    x: 21,
    y: null,
    key: '2022-05-21',
    name: '2022-05-21',
    units: undefined,
  },
  {
    x: 22,
    y: null,
    key: '2022-05-22',
    name: '2022-05-22',
    units: undefined,
  },
  {
    x: 23,
    y: null,
    key: '2022-05-23',
    name: '2022-05-23',
    units: undefined,
  },
  {
    x: 24,
    y: null,
    key: '2022-05-24',
    name: '2022-05-24',
    units: undefined,
  },
  {
    x: 25,
    y: null,
    key: '2022-05-25',
    name: '2022-05-25',
    units: undefined,
  },
  {
    x: 26,
    y: null,
    key: '2022-05-26',
    name: '2022-05-26',
    units: undefined,
  },
  {
    x: 27,
    y: null,
    key: '2022-05-27',
    name: '2022-05-27',
    units: undefined,
  },
  {
    x: 28,
    y: null,
    key: '2022-05-28',
    name: '2022-05-28',
    units: undefined,
  },
  {
    x: 29,
    y: null,
    key: '2022-05-29',
    name: '2022-05-29',
    units: undefined,
  },
  {
    x: 30,
    y: null,
    key: '2022-05-30',
    name: '2022-05-30',
    units: undefined,
  },
  {
    x: 31,
    y: null,
    key: '2022-05-31',
    name: '2022-05-31',
    units: undefined,
  },
];

export { transformReportReturns };
