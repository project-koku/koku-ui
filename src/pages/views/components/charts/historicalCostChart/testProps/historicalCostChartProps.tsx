const HistoricalCostChartTestProps = {
  currentCostData: [
    {
      x: 1,
      y: 464.63,
      key: '2022-05-01',
      name: '2022-05-01',
      units: 'USD',
    },
    {
      x: 2,
      y: 584.16,
      key: '2022-05-02',
      name: '2022-05-02',
      units: 'USD',
    },
    {
      x: 3,
      y: 703.71,
      key: '2022-05-03',
      name: '2022-05-03',
      units: 'USD',
    },
    {
      x: 4,
      y: 823.26,
      key: '2022-05-04',
      name: '2022-05-04',
      units: 'USD',
    },
    {
      x: 5,
      y: 942.79,
      key: '2022-05-05',
      name: '2022-05-05',
      units: 'USD',
    },
    {
      x: 6,
      y: 1062.34,
      key: '2022-05-06',
      name: '2022-05-06',
      units: 'USD',
    },
    {
      x: 7,
      y: 1181.89,
      key: '2022-05-07',
      name: '2022-05-07',
      units: 'USD',
    },
    {
      x: 8,
      y: 1301.43,
      key: '2022-05-08',
      name: '2022-05-08',
      units: 'USD',
    },
    {
      x: 9,
      y: 1420.97,
      key: '2022-05-09',
      name: '2022-05-09',
      units: 'USD',
    },
    {
      x: 10,
      y: 1540.52,
      key: '2022-05-10',
      name: '2022-05-10',
      units: 'USD',
    },
    {
      x: 11,
      y: 1660.07,
      key: '2022-05-11',
      name: '2022-05-11',
      units: 'USD',
    },
    {
      x: 12,
      y: 1825.69,
      key: '2022-05-12',
      name: '2022-05-12',
      units: 'USD',
    },
    {
      x: 13,
      y: 2057.52,
      key: '2022-05-13',
      name: '2022-05-13',
      units: 'USD',
    },
    {
      x: 14,
      y: 2288.33,
      key: '2022-05-14',
      name: '2022-05-14',
      units: 'USD',
    },
    {
      x: 15,
      y: 2514.21,
      key: '2022-05-15',
      name: '2022-05-15',
      units: 'USD',
    },
    {
      x: 16,
      y: 2747.85,
      key: '2022-05-16',
      name: '2022-05-16',
      units: 'USD',
    },
    {
      x: 17,
      y: 2973.13,
      key: '2022-05-17',
      name: '2022-05-17',
      units: 'USD',
    },
    {
      x: 18,
      y: null,
      key: '2022-05-18',
      name: '2022-05-18',
    },
    {
      x: 19,
      y: null,
      key: '2022-05-19',
      name: '2022-05-19',
    },
    {
      x: 20,
      y: null,
      key: '2022-05-20',
      name: '2022-05-20',
    },
    {
      x: 21,
      y: null,
      key: '2022-05-21',
      name: '2022-05-21',
    },
    {
      x: 22,
      y: null,
      key: '2022-05-22',
      name: '2022-05-22',
    },
    {
      x: 23,
      y: null,
      key: '2022-05-23',
      name: '2022-05-23',
    },
    {
      x: 24,
      y: null,
      key: '2022-05-24',
      name: '2022-05-24',
    },
    {
      x: 25,
      y: null,
      key: '2022-05-25',
      name: '2022-05-25',
    },
    {
      x: 26,
      y: null,
      key: '2022-05-26',
      name: '2022-05-26',
    },
    {
      x: 27,
      y: null,
      key: '2022-05-27',
      name: '2022-05-27',
    },
    {
      x: 28,
      y: null,
      key: '2022-05-28',
      name: '2022-05-28',
    },
    {
      x: 29,
      y: null,
      key: '2022-05-29',
      name: '2022-05-29',
    },
    {
      x: 30,
      y: null,
      key: '2022-05-30',
      name: '2022-05-30',
    },
    {
      x: 31,
      y: null,
      key: '2022-05-31',
      name: '2022-05-31',
    },
  ],
  currentInfrastructureCostData: [
    {
      x: 1,
      y: 46.76,
      key: '2022-05-01',
      name: '2022-05-01',
      units: 'USD',
    },
    {
      x: 2,
      y: 166.29,
      key: '2022-05-02',
      name: '2022-05-02',
      units: 'USD',
    },
    {
      x: 3,
      y: 285.84,
      key: '2022-05-03',
      name: '2022-05-03',
      units: 'USD',
    },
    {
      x: 4,
      y: 405.39,
      key: '2022-05-04',
      name: '2022-05-04',
      units: 'USD',
    },
    {
      x: 5,
      y: 524.92,
      key: '2022-05-05',
      name: '2022-05-05',
      units: 'USD',
    },
    {
      x: 6,
      y: 644.47,
      key: '2022-05-06',
      name: '2022-05-06',
      units: 'USD',
    },
    {
      x: 7,
      y: 764.02,
      key: '2022-05-07',
      name: '2022-05-07',
      units: 'USD',
    },
    {
      x: 8,
      y: 883.56,
      key: '2022-05-08',
      name: '2022-05-08',
      units: 'USD',
    },
    {
      x: 9,
      y: 1003.1,
      key: '2022-05-09',
      name: '2022-05-09',
      units: 'USD',
    },
    {
      x: 10,
      y: 1122.65,
      key: '2022-05-10',
      name: '2022-05-10',
      units: 'USD',
    },
    {
      x: 11,
      y: 1242.2,
      key: '2022-05-11',
      name: '2022-05-11',
      units: 'USD',
    },
    {
      x: 12,
      y: 1366.92,
      key: '2022-05-12',
      name: '2022-05-12',
      units: 'USD',
    },
    {
      x: 13,
      y: 1491.31,
      key: '2022-05-13',
      name: '2022-05-13',
      units: 'USD',
    },
    {
      x: 14,
      y: 1615.57,
      key: '2022-05-14',
      name: '2022-05-14',
      units: 'USD',
    },
    {
      x: 15,
      y: 1740.04,
      key: '2022-05-15',
      name: '2022-05-15',
      units: 'USD',
    },
    {
      x: 16,
      y: 1864.46,
      key: '2022-05-16',
      name: '2022-05-16',
      units: 'USD',
    },
    {
      x: 17,
      y: 1982.76,
      key: '2022-05-17',
      name: '2022-05-17',
      units: 'USD',
    },
    {
      x: 18,
      y: null,
      key: '2022-05-18',
      name: '2022-05-18',
    },
    {
      x: 19,
      y: null,
      key: '2022-05-19',
      name: '2022-05-19',
    },
    {
      x: 20,
      y: null,
      key: '2022-05-20',
      name: '2022-05-20',
    },
    {
      x: 21,
      y: null,
      key: '2022-05-21',
      name: '2022-05-21',
    },
    {
      x: 22,
      y: null,
      key: '2022-05-22',
      name: '2022-05-22',
    },
    {
      x: 23,
      y: null,
      key: '2022-05-23',
      name: '2022-05-23',
    },
    {
      x: 24,
      y: null,
      key: '2022-05-24',
      name: '2022-05-24',
    },
    {
      x: 25,
      y: null,
      key: '2022-05-25',
      name: '2022-05-25',
    },
    {
      x: 26,
      y: null,
      key: '2022-05-26',
      name: '2022-05-26',
    },
    {
      x: 27,
      y: null,
      key: '2022-05-27',
      name: '2022-05-27',
    },
    {
      x: 28,
      y: null,
      key: '2022-05-28',
      name: '2022-05-28',
    },
    {
      x: 29,
      y: null,
      key: '2022-05-29',
      name: '2022-05-29',
    },
    {
      x: 30,
      y: null,
      key: '2022-05-30',
      name: '2022-05-30',
    },
    {
      x: 31,
      y: null,
      key: '2022-05-31',
      name: '2022-05-31',
    },
  ],
  previousCostData: [
    {
      x: 1,
      y: 355.77,
      key: '2022-04-01',
      name: '2022-04-01',
      units: 'USD',
    },
    {
      x: 2,
      y: 475.31,
      key: '2022-04-02',
      name: '2022-04-02',
      units: 'USD',
    },
    {
      x: 3,
      y: 594.82,
      key: '2022-04-03',
      name: '2022-04-03',
      units: 'USD',
    },
    {
      x: 4,
      y: 713.12,
      key: '2022-04-04',
      name: '2022-04-04',
      units: 'USD',
    },
    {
      x: 5,
      y: 786.35,
      key: '2022-04-05',
      name: '2022-04-05',
      units: 'USD',
    },
    {
      x: 6,
      y: 905.9,
      key: '2022-04-06',
      name: '2022-04-06',
      units: 'USD',
    },
    {
      x: 7,
      y: 1025.44,
      key: '2022-04-07',
      name: '2022-04-07',
      units: 'USD',
    },
    {
      x: 8,
      y: 1144.97,
      key: '2022-04-08',
      name: '2022-04-08',
      units: 'USD',
    },
    {
      x: 9,
      y: 1264.51,
      key: '2022-04-09',
      name: '2022-04-09',
      units: 'USD',
    },
    {
      x: 10,
      y: 1384.05,
      key: '2022-04-10',
      name: '2022-04-10',
      units: 'USD',
    },
    {
      x: 11,
      y: 1503.58,
      key: '2022-04-11',
      name: '2022-04-11',
      units: 'USD',
    },
    {
      x: 12,
      y: 1623.11,
      key: '2022-04-12',
      name: '2022-04-12',
      units: 'USD',
    },
    {
      x: 13,
      y: 1742.64,
      key: '2022-04-13',
      name: '2022-04-13',
      units: 'USD',
    },
    {
      x: 14,
      y: 1862.17,
      key: '2022-04-14',
      name: '2022-04-14',
      units: 'USD',
    },
    {
      x: 15,
      y: 1981.72,
      key: '2022-04-15',
      name: '2022-04-15',
      units: 'USD',
    },
    {
      x: 16,
      y: 2101.27,
      key: '2022-04-16',
      name: '2022-04-16',
      units: 'USD',
    },
    {
      x: 17,
      y: 2220.81,
      key: '2022-04-17',
      name: '2022-04-17',
      units: 'USD',
    },
    {
      x: 18,
      y: 2340.33,
      key: '2022-04-18',
      name: '2022-04-18',
      units: 'USD',
    },
    {
      x: 19,
      y: 2459.86,
      key: '2022-04-19',
      name: '2022-04-19',
      units: 'USD',
    },
    {
      x: 20,
      y: 2579.39,
      key: '2022-04-20',
      name: '2022-04-20',
      units: 'USD',
    },
    {
      x: 21,
      y: 2698.91,
      key: '2022-04-21',
      name: '2022-04-21',
      units: 'USD',
    },
    {
      x: 22,
      y: 2818.43,
      key: '2022-04-22',
      name: '2022-04-22',
      units: 'USD',
    },
    {
      x: 23,
      y: 2937.95,
      key: '2022-04-23',
      name: '2022-04-23',
      units: 'USD',
    },
    {
      x: 24,
      y: 3057.49,
      key: '2022-04-24',
      name: '2022-04-24',
      units: 'USD',
    },
    {
      x: 25,
      y: 3058.73,
      key: '2022-04-25',
      name: '2022-04-25',
      units: 'USD',
    },
    {
      x: 26,
      y: 3059.97,
      key: '2022-04-26',
      name: '2022-04-26',
      units: 'USD',
    },
    {
      x: 27,
      y: 3061.22,
      key: '2022-04-27',
      name: '2022-04-27',
      units: 'USD',
    },
    {
      x: 28,
      y: 3180.75,
      key: '2022-04-28',
      name: '2022-04-28',
      units: 'USD',
    },
    {
      x: 29,
      y: 3300.28,
      key: '2022-04-29',
      name: '2022-04-29',
      units: 'USD',
    },
    {
      x: 30,
      y: 3419.83,
      key: '2022-04-30',
      name: '2022-04-30',
      units: 'USD',
    },
  ],
  previousInfrastructureCostData: [
    {
      x: 1,
      y: 46.74,
      key: '2022-04-01',
      name: '2022-04-01',
      units: 'USD',
    },
    {
      x: 2,
      y: 166.28,
      key: '2022-04-02',
      name: '2022-04-02',
      units: 'USD',
    },
    {
      x: 3,
      y: 285.79,
      key: '2022-04-03',
      name: '2022-04-03',
      units: 'USD',
    },
    {
      x: 4,
      y: 404.09,
      key: '2022-04-04',
      name: '2022-04-04',
      units: 'USD',
    },
    {
      x: 5,
      y: 477.32,
      key: '2022-04-05',
      name: '2022-04-05',
      units: 'USD',
    },
    {
      x: 6,
      y: 596.87,
      key: '2022-04-06',
      name: '2022-04-06',
      units: 'USD',
    },
    {
      x: 7,
      y: 716.41,
      key: '2022-04-07',
      name: '2022-04-07',
      units: 'USD',
    },
    {
      x: 8,
      y: 835.94,
      key: '2022-04-08',
      name: '2022-04-08',
      units: 'USD',
    },
    {
      x: 9,
      y: 955.48,
      key: '2022-04-09',
      name: '2022-04-09',
      units: 'USD',
    },
    {
      x: 10,
      y: 1075.02,
      key: '2022-04-10',
      name: '2022-04-10',
      units: 'USD',
    },
    {
      x: 11,
      y: 1194.55,
      key: '2022-04-11',
      name: '2022-04-11',
      units: 'USD',
    },
    {
      x: 12,
      y: 1314.08,
      key: '2022-04-12',
      name: '2022-04-12',
      units: 'USD',
    },
    {
      x: 13,
      y: 1433.61,
      key: '2022-04-13',
      name: '2022-04-13',
      units: 'USD',
    },
    {
      x: 14,
      y: 1553.14,
      key: '2022-04-14',
      name: '2022-04-14',
      units: 'USD',
    },
    {
      x: 15,
      y: 1672.69,
      key: '2022-04-15',
      name: '2022-04-15',
      units: 'USD',
    },
    {
      x: 16,
      y: 1792.24,
      key: '2022-04-16',
      name: '2022-04-16',
      units: 'USD',
    },
    {
      x: 17,
      y: 1911.78,
      key: '2022-04-17',
      name: '2022-04-17',
      units: 'USD',
    },
    {
      x: 18,
      y: 2031.3,
      key: '2022-04-18',
      name: '2022-04-18',
      units: 'USD',
    },
    {
      x: 19,
      y: 2150.83,
      key: '2022-04-19',
      name: '2022-04-19',
      units: 'USD',
    },
    {
      x: 20,
      y: 2270.36,
      key: '2022-04-20',
      name: '2022-04-20',
      units: 'USD',
    },
    {
      x: 21,
      y: 2389.88,
      key: '2022-04-21',
      name: '2022-04-21',
      units: 'USD',
    },
    {
      x: 22,
      y: 2509.4,
      key: '2022-04-22',
      name: '2022-04-22',
      units: 'USD',
    },
    {
      x: 23,
      y: 2628.92,
      key: '2022-04-23',
      name: '2022-04-23',
      units: 'USD',
    },
    {
      x: 24,
      y: 2748.46,
      key: '2022-04-24',
      name: '2022-04-24',
      units: 'USD',
    },
    {
      x: 25,
      y: 2749.7,
      key: '2022-04-25',
      name: '2022-04-25',
      units: 'USD',
    },
    {
      x: 26,
      y: 2750.94,
      key: '2022-04-26',
      name: '2022-04-26',
      units: 'USD',
    },
    {
      x: 27,
      y: 2752.19,
      key: '2022-04-27',
      name: '2022-04-27',
      units: 'USD',
    },
    {
      x: 28,
      y: 2871.72,
      key: '2022-04-28',
      name: '2022-04-28',
      units: 'USD',
    },
    {
      x: 29,
      y: 2991.25,
      key: '2022-04-29',
      name: '2022-04-29',
      units: 'USD',
    },
    {
      x: 30,
      y: 3110.8,
      key: '2022-04-30',
      name: '2022-04-30',
      units: 'USD',
    },
  ],
  xAxisLabel: 'Day of Month',
  yAxisLabel: 'Cost ($)',
};
export { HistoricalCostChartTestProps };
