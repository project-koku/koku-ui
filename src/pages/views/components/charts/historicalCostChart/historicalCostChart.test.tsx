import { render, screen } from '@testing-library/react';
import React from 'react';

import { HistoricalCostChart, HistoricalCostChartProps } from './historicalCostChart';
import { HistoricalCostChartTestProps } from './testProps/historicalCostChartProps';

const props: HistoricalCostChartProps = {
  ...HistoricalCostChartTestProps,
  intl: {
    formatMessage: jest.fn((m, v) => JSON.stringify(v)),
  } as any,
  height: 100,
  title: 'Usage Title',
  formatter: jest.fn(),
  formatOptions: {},
};

test('reports are propertly generated', () => {
  const view = render(<HistoricalCostChart {...props} />);

  // i'd like to do this (test legend), but i'm having trouble mocking intl unless it's a direct property
  // expect(screen.getByText(/Cost \(Apr 1-30\)/i)).not.toBeNull();
  // expect(screen.getByText(/Cost \(May 1-17\)/i)).not.toBeNull();
  // expect(screen.getByText(/Infrastructure cost \(Apr 1-30\)/i)).not.toBeNull();
  // expect(screen.getByText(/Infrastructure cost \(May 1-17\)/i)).not.toBeNull();

  // below is to capture all the graph points which are contained within an svg
  expect(view.container).toMatchSnapshot();
});

test('null previous and current reports are handled', () => {
  const view = render(
    <HistoricalCostChart
      {...props}
      currentCostData={null}
      currentInfrastructureCostData={null}
      previousCostData={null}
      previousInfrastructureCostData={null}
    />
  );
  const linesWithNoData = screen.getAllByText(/no data/i);
  expect(linesWithNoData.length).toBe(4);
  // below is to capture all the graph points which are contained within an svg
  expect(view.container).toMatchSnapshot();
});

test('height from props is used', () => {
  render(<HistoricalCostChart {...props} />);
  expect(screen.getByTestId('historical-chart-wrapper').getAttribute('style')).toContain('height: 100px');
});
