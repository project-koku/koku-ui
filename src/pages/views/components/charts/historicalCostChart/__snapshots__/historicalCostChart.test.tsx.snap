// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reports are formatted to datums 1`] = `
<g>
  <rect
    height="61.809999999999995"
    role="presentation"
    shape-rendering="auto"
    style="fill: none;"
    vector-effect="non-scaling-stroke"
    width="18134.518750000003"
    x="0"
    y="68"
  />
  <path
    d="M 7.84, 84.156
      h12.32
      v-4.312
      h-12.32
      z"
    role="presentation"
    shape-rendering="auto"
    style="fill: #8bc1f7;"
  />
  <path
    d="M 9075.099375000002, 84.156
      h12.319999999999709
      v-4.312
      h-12.319999999999709
      z"
    role="presentation"
    shape-rendering="auto"
    style="fill: #38812f;"
  />
  <path
    d="M 7.84, 115.061
      h3.6959999999999997
      v-4.312
      h-3.6959999999999997
      z
      M 12.767999999999999, 115.061
      h3.6959999999999997
      v-4.312
      h-3.6959999999999997
      z
      M 17.695999999999998, 115.061
      h3.6959999999999997
      v-4.312
      h-3.6959999999999997
      z"
    role="presentation"
    shape-rendering="auto"
    style="fill: #519de9;"
  />
  <path
    d="M 9075.099375000002, 115.061
      h3.6959999999999127
      v-4.312
      h-3.6959999999999127
      z
      M 9080.027375000001, 115.061
      h3.6959999999999127
      v-4.312
      h-3.6959999999999127
      z
      M 9084.955375000001, 115.061
      h3.6959999999999127
      v-4.312
      h-3.6959999999999127
      z"
    role="presentation"
    shape-rendering="auto"
    style="fill: #4cb140;"
  />
  <text
    direction="inherit"
    dx="0"
    id="legend-labels-0"
    x="30.8"
    y="86.97"
  >
    <tspan
      dx="0"
      dy="0"
      style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #252525;"
      text-anchor="start"
      x="30.8"
    >
      {month, select, 0 {{count, plural, one {Cost (Jan {startDate})} other {Cost (Jan {startDate}-{endDate})}}} 1 {{count, plural, one {Cost (Feb {startDate})} other {Cost (Feb {startDate}-{endDate})}}} 2 {{count, plural, one {Cost (Mar {startDate})} other {Cost (Mar {startDate}-{endDate})}}} 3 {{count, plural, one {Cost (Apr {startDate})} other {Cost (Apr {startDate}-{endDate})}}} 4 {{count, plural, one {Cost (May {startDate})} other {Cost (May {startDate}-{endDate})}}} 5 {{count, plural, one {Cost (Jun {startDate})} other {Cost (Jun {startDate}-{endDate})}}} 6 {{count, plural, one {Cost (Jul {startDate})} other {Cost (Jul {startDate}-{endDate})}}} 7 {{count, plural, one {Cost (Aug {startDate})} other {Cost (Aug {startDate}-{endDate})}}} 8 {{count, plural, one {Cost (Sep {startDate})} other {Cost (Sep {startDate}-{endDate})}}} 9 {{count, plural, one {Cost (Oct {startDate})} other {Cost (Oct {startDate}-{endDate})}}} 10 {{count, plural, one {Cost (Nov {startDate})} other {Cost (Nov {startDate}-{endDate})}}} 11 {{count, plural, one {Cost (Dec {startDate})} other {Cost (Dec {startDate}-{endDate})}}} other {}}
    </tspan>
  </text>
  <text
    direction="inherit"
    dx="0"
    id="legend-labels-1"
    x="9098.059375"
    y="86.97"
  >
    <tspan
      dx="0"
      dy="0"
      style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #252525;"
      text-anchor="start"
      x="9098.059375"
    >
      {month, select, 0 {{count, plural, one {Cost (Jan {startDate})} other {Cost (Jan {startDate}-{endDate})}}} 1 {{count, plural, one {Cost (Feb {startDate})} other {Cost (Feb {startDate}-{endDate})}}} 2 {{count, plural, one {Cost (Mar {startDate})} other {Cost (Mar {startDate}-{endDate})}}} 3 {{count, plural, one {Cost (Apr {startDate})} other {Cost (Apr {startDate}-{endDate})}}} 4 {{count, plural, one {Cost (May {startDate})} other {Cost (May {startDate}-{endDate})}}} 5 {{count, plural, one {Cost (Jun {startDate})} other {Cost (Jun {startDate}-{endDate})}}} 6 {{count, plural, one {Cost (Jul {startDate})} other {Cost (Jul {startDate}-{endDate})}}} 7 {{count, plural, one {Cost (Aug {startDate})} other {Cost (Aug {startDate}-{endDate})}}} 8 {{count, plural, one {Cost (Sep {startDate})} other {Cost (Sep {startDate}-{endDate})}}} 9 {{count, plural, one {Cost (Oct {startDate})} other {Cost (Oct {startDate}-{endDate})}}} 10 {{count, plural, one {Cost (Nov {startDate})} other {Cost (Nov {startDate}-{endDate})}}} 11 {{count, plural, one {Cost (Dec {startDate})} other {Cost (Dec {startDate}-{endDate})}}} other {}}
    </tspan>
  </text>
  <text
    direction="inherit"
    dx="0"
    id="legend-labels-2"
    x="30.8"
    y="117.875"
  >
    <tspan
      dx="0"
      dy="0"
      style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #252525;"
      text-anchor="start"
      x="30.8"
    >
      {month, select, 0 {{count, plural, one {Infrastructure cost (Jan {startDate})} other {Infrastructure cost (Jan {startDate}-{endDate})}}} 1 {{count, plural, one {Infrastructure cost (Feb {startDate})} other {Infrastructure cost (Feb {startDate}-{endDate})}}} 2 {{count, plural, one {Infrastructure cost (Mar {startDate})} other {Infrastructure cost (Mar {startDate}-{endDate})}}} 3 {{count, plural, one {Infrastructure cost (Apr {startDate})} other {Infrastructure cost (Apr {startDate}-{endDate})}}} 4 {{count, plural, one {Infrastructure cost (May {startDate})} other {Infrastructure cost (May {startDate}-{endDate})}}} 5 {{count, plural, one {Infrastructure cost (Jun {startDate})} other {Infrastructure cost (Jun {startDate}-{endDate})}}} 6 {{count, plural, one {Infrastructure cost (Jul {startDate})} other {Infrastructure cost (Jul {startDate}-{endDate})}}} 7 {{count, plural, one {Infrastructure cost (Aug {startDate})} other {Infrastructure cost (Aug {startDate}-{endDate})}}} 8 {{count, plural, one {Infrastructure cost (Sep {startDate})} other {Infrastructure cost (Sep {startDate}-{endDate})}}} 9 {{count, plural, one {Infrastructure cost (Oct {startDate})} other {Infrastructure cost (Oct {startDate}-{endDate})}}} 10 {{count, plural, one {Infrastructure cost (Nov {startDate})} other {Infrastructure cost (Nov {startDate}-{endDate})}}} 11 {{count, plural, one {Infrastructure cost (Dec {startDate})} other {Infrastructure cost (Dec {startDate}-{endDate})}}} other {}}
    </tspan>
  </text>
  <text
    direction="inherit"
    dx="0"
    id="legend-labels-3"
    x="9098.059375"
    y="117.875"
  >
    <tspan
      dx="0"
      dy="0"
      style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #252525;"
      text-anchor="start"
      x="9098.059375"
    >
      {month, select, 0 {{count, plural, one {Infrastructure cost (Jan {startDate})} other {Infrastructure cost (Jan {startDate}-{endDate})}}} 1 {{count, plural, one {Infrastructure cost (Feb {startDate})} other {Infrastructure cost (Feb {startDate}-{endDate})}}} 2 {{count, plural, one {Infrastructure cost (Mar {startDate})} other {Infrastructure cost (Mar {startDate}-{endDate})}}} 3 {{count, plural, one {Infrastructure cost (Apr {startDate})} other {Infrastructure cost (Apr {startDate}-{endDate})}}} 4 {{count, plural, one {Infrastructure cost (May {startDate})} other {Infrastructure cost (May {startDate}-{endDate})}}} 5 {{count, plural, one {Infrastructure cost (Jun {startDate})} other {Infrastructure cost (Jun {startDate}-{endDate})}}} 6 {{count, plural, one {Infrastructure cost (Jul {startDate})} other {Infrastructure cost (Jul {startDate}-{endDate})}}} 7 {{count, plural, one {Infrastructure cost (Aug {startDate})} other {Infrastructure cost (Aug {startDate}-{endDate})}}} 8 {{count, plural, one {Infrastructure cost (Sep {startDate})} other {Infrastructure cost (Sep {startDate}-{endDate})}}} 9 {{count, plural, one {Infrastructure cost (Oct {startDate})} other {Infrastructure cost (Oct {startDate}-{endDate})}}} 10 {{count, plural, one {Infrastructure cost (Nov {startDate})} other {Infrastructure cost (Nov {startDate}-{endDate})}}} 11 {{count, plural, one {Infrastructure cost (Dec {startDate})} other {Infrastructure cost (Dec {startDate}-{endDate})}}} other {}}
    </tspan>
  </text>
</g>
`;
